pkg/build/api/types.go

```diff
@@ -412,6 +416,89 @@ type SourceBuildStrategy struct {
 	ForcePull bool
 }
 
+// A BuildPostCommitSpec holds a build post commit hook specification. The hook
+// executes a command in a temporary container running the build output image,
+// immediately after the last layer of the image is commited and before the
+// image is pushed to a registry. The command is executed with the current
+// working directory ($PWD) set to the image's WORKDIR.
+//
+// The build will be marked as failed if the hook execution fails. It will fail
+// if the script or command return a non-zero exit code, or if there is any
+// other error related to starting the temporary container.
+//
+// There are five different ways to configure the hook. As an example, all forms
+// below are equivalent and will execute `rake test --verbose`.
+//
+// 1. Shell script:
+//
+// 	BuildPostCommitSpec{
+// 		Script: "rake test --verbose",
+// 	}
+//
+// The above is a convenient form which is equivalent to:
+//
+// 	BuildPostCommitSpec{
+// 		Command: []string{"/bin/sh", "-c"},
+// 		Args: []string{"rake test --verbose"},
+// 	}
+//
+// 2. Command as the image entrypoint:
+//
+// 	BuildPostCommitSpec{
+// 		Command: []string{"rake", "test", "--verbose"},
+// 	}
+//
+// Command overrides the image entrypoint in the exec form, as documented in
+// Docker: https://docs.docker.com/engine/reference/builder/#entrypoint.
+//
+// 3. Pass arguments to the default entrypoint:
+//
+// 	BuildPostCommitSpec{
+// 		Args: []string{"rake", "test", "--verbose"},
+// 	}
+//
+// This form is only useful if the image entrypoint can handle arguments.
+//
+// 4. Shell script with arguments:
+//
+// 	BuildPostCommitSpec{
+// 		Script: "rake test $1",
+// 		Args: []string{"--verbose"},
+// 	}
+//
+// This form is useful if you need to pass arguments that would otherwise be
+// hard to quote properly in the shell script. In the script, $0 will be
+// "/bin/sh" and $1, $2, etc, are the positional arguments from Args.
+//
+// 5. Command with arguments:
+//
+// 	BuildPostCommitSpec{
+// 		Command: []string{"rake", "test"},
+// 		Args: []string{"--verbose"},
+// 	}
+//
+// This form is equivalent to appending the arguments to the Command slice.
+//
+// It is invalid to provide both Script and Command simultaneously. If none of
+// the fields are specified, the hook is not executed.
+type BuildPostCommitSpec struct {
```

Comment:
Two comments, otherwise this looks good and you have API approval.

On Thu, Feb 4, 2016 at 8:50 AM, Rodolfo Carvalho <notifications@github.com>
wrote:

> In pkg/build/api/types.go
> <https://github.com/openshift/origin/pull/6715#discussion_r51873695>:
>
> > +// This form is useful if you need to pass arguments that would otherwise be
> > +// hard to quote properly in the shell script. In the script, $0 will be
> > +// "/bin/sh" and $1, $2, etc, are the positional arguments from Args.
> > +//
> > +// 5. Command with arguments:
> > +//
> > +// 	BuildPostCommitSpec{
> > +// 		Command: []string{"rake", "test"},
> > +// 		Args: []string{"--verbose"},
> > +// 	}
> > +//
> > +// This form is equivalent to appending the arguments to the Command slice.
> > +//
> > +// It is invalid to provide both Script and Command simultaneously. If none of
> > +// the fields are specified, the hook is not executed.
> > +type BuildPostCommitSpec struct {
>
> @smarterclayton <https://github.com/smarterclayton> could you please have
> another look here? Me & @PI-Victor <https://github.com/PI-Victor> rewrote
> the docs. Depending on your feedback we'll update v1/types and
> v1beta3/types.go
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/6715/files#r51873695>.
>